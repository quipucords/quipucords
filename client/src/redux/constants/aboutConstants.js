const ABOUT_DIALOG_OPEN = 'ABOUT_DIALOG_OPEN';
const ABOUT_DIALOG_CLOSE = 'ABOUT_DIALOG_CLOSE';

export { ABOUT_DIALOG_OPEN, ABOUT_DIALOG_CLOSE };
