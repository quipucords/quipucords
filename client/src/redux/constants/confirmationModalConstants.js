const CONFIRMATION_MODAL_SHOW = 'CONFIRMATION_MODAL_SHOW';
const CONFIRMATION_MODAL_HIDE = 'CONFIRMATION_MODAL_HIDE';

export { CONFIRMATION_MODAL_SHOW, CONFIRMATION_MODAL_HIDE };
