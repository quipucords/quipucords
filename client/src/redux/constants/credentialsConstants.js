
const LOAD_CREDENTIALS_SUCCESS = 'LOAD_CREDENTIALS_SUCCESS';

export {
  LOAD_CREDENTIALS_SUCCESS
}
