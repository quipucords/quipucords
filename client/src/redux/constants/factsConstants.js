const ADD_FACTS_ERROR = 'ADD_FACTS_ERROR';
const ADD_FACTS_LOADING = 'ADD_FACTS_LOADING';
const ADD_FACTS_SUCCESS = 'ADD_FACTS_SUCCESS';

export { ADD_FACTS_ERROR, ADD_FACTS_LOADING, ADD_FACTS_SUCCESS };
