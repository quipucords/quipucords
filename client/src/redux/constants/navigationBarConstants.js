const NAV_TOGGLE_COLLAPSE = 'NAV_TOGGLE_COLLAPSE';
const NAV_COLLAPSE = 'NAV_COLLAPSE';
const NAV_OPEN = 'NAV_OPEN';

export { NAV_TOGGLE_COLLAPSE, NAV_COLLAPSE, NAV_OPEN };
