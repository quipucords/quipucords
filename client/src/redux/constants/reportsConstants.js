const GET_REPORT = 'GET_REPORT';
const GET_MERGE_REPORT = 'GET_MERGE_REPORT';

export { GET_REPORT, GET_MERGE_REPORT };
