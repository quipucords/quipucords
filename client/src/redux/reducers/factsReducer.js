import helpers from '../../common/helpers';
import { factsTypes } from '../constants';

const initialState = {
  update: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    facts: {}
  }
};

const factsReducer = (state = initialState, action) => {
  switch (action.type) {
    case helpers.REJECTED_ACTION(factsTypes.ADD_FACTS):
      return helpers.setStateProp(
        'update',
        {
          error: action.error,
          errorMessage: helpers.getMessageFromResults(action.payload).message
        },
        {
          state,
          initialState
        }
      );

    case helpers.PENDING_ACTION(factsTypes.ADD_FACTS):
      return helpers.setStateProp(
        'update',
        {
          pending: true
        },
        {
          state,
          initialState
        }
      );

    case helpers.FULFILLED_ACTION(factsTypes.ADD_FACTS):
      return helpers.setStateProp(
        'update',
        {
          facts: action.payload.data,
          fulfilled: true
        },
        {
          state,
          initialState
        }
      );

    default:
      return state;
  }
};

factsReducer.initialState = initialState;

export { factsReducer as default, initialState, factsReducer };
