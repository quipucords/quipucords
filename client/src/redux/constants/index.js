import * as aboutTypes from './aboutConstants';
import * as credentialsTypes from './credentialsConstants';
import * as factsTypes from './factsConstants';
import * as navigationBarTypes from './navigationBarConstants';
import * as reportsTypes from './reportsConstants';
import * as scansTypes from './scansConstants';
import * as sourcesTypes from './sourcesConstants';
import * as viewToolbarTypes from './viewToolbarConstants';

const reduxTypes = {
  about: aboutTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  navigation: navigationBarTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes,
  aboutTypes,
  credentialsTypes,
  factsTypes,
  navigationBarTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  viewToolbarTypes
};

export default reduxTypes;
