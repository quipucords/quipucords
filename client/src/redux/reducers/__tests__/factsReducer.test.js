import helpers from '../../../common/helpers';
import { factsTypes } from '../../constants/index';
import factsReducer from '../factsReducer';

const initialState = {
  persist: {},

  update: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    facts: {}
  }
};

describe('factsReducer', function() {
  it('should return the initial state', () => {
    expect(factsReducer(undefined, {})).toEqual(initialState);
  });

  it('should handle ADD_FACTS_REJECTED', () => {
    let dispatched = {
      type: helpers.REJECTED_ACTION(factsTypes.ADD_FACTS),
      error: true,
      payload: {
        message: 'BACKUP MESSAGE',
        response: {
          data: {
            detail: 'GET ERROR'
          }
        }
      }
    };

    let resultState = factsReducer(undefined, dispatched);
    expect(resultState.update.error).toBeTruthy();
    expect(resultState.update.errorMessage).toEqual('GET ERROR');

    expect(resultState.persist).toEqual(initialState.persist);
  });

  it('should handle ADD_FACTS_PENDING', () => {
    let dispatched = {
      type: helpers.PENDING_ACTION(factsTypes.ADD_FACTS)
    };

    let resultState = factsReducer(undefined, dispatched);

    expect(resultState.update.pending).toBeTruthy();
    expect(resultState.persist).toEqual(initialState.persist);
  });

  it('should handle ADD_FACTS_FULFILLED', () => {
    let dispatched = {
      type: helpers.FULFILLED_ACTION(factsTypes.ADD_FACTS),
      payload: {
        data: [
          {
            name: '1',
            id: 1
          },
          {
            name: '2',
            id: 2
          },
          {
            name: '3',
            id: 3
          },
          {
            name: '4',
            id: 4
          }
        ]
      }
    };

    let resultState = factsReducer(undefined, dispatched);

    expect(resultState.update.fulfilled).toBeTruthy();
    expect(resultState.update.facts).toHaveLength(4);

    expect(resultState.persist).toEqual(initialState.persist);
  });
});
