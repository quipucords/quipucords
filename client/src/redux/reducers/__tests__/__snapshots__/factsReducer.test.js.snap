// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fulfilled types ADD_FACTS 1`] = `
Object {
  "result": Object {
    "update": Object {
      "error": false,
      "errorMessage": "",
      "facts": Object {
        "test": "success",
      },
      "fulfilled": true,
      "pending": false,
    },
  },
  "type": "ADD_FACTS_FULFILLED",
}
`;

exports[`pending types ADD_FACTS 1`] = `
Object {
  "result": Object {
    "update": Object {
      "error": false,
      "errorMessage": "",
      "facts": Object {},
      "fulfilled": false,
      "pending": true,
    },
  },
  "type": "ADD_FACTS_PENDING",
}
`;

exports[`rejected types ADD_FACTS 1`] = `
Object {
  "result": Object {
    "update": Object {
      "error": true,
      "errorMessage": "ERROR",
      "facts": Object {},
      "fulfilled": false,
      "pending": false,
    },
  },
  "type": "ADD_FACTS_REJECTED",
}
`;
